bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -3031,18 +3031,21 @@ private void shutdown() {
 		registry.removeRegistryChangeListener(extensionEventHandler);
 		registry.removeRegistryChangeListener(startupRegistryListener);
 
+		// shutdown activity helper before disposing workbench activity support;
+		// dispose activity support before disposing service locator to avoid
+		// unnecessary activity disablement processing
```

Comment:
Would it be possible to test this? If I'm reading it correctly, this is the cornerstone of the PR, right? I mean the order is relevant.

Reply:
Thanks @fedejeanne, that's a good idea. I would actually not like to test the execution order, as it should be an implementation detail of the method. Instead, I would propose to test the effect that I want to achieve or avoid, respectively. In this case, This PR aims at avoiding unnecessary activity manager events being fired during workbench shutdown, So I've added a test that verifies this behavior in 26964bb. The test fails when executed on the code without the changes in this PR, so it's actually a regression test for the according changes.

Reply:
Good idea! And you're right: the less one tests the implementation details, the more stable the tests are.

Thanks for adding the regression test.